src/coreComponents/codingUtilities/Table.hpp

```diff
@@ -0,0 +1,224 @@
+   * @param largestHeaderVectorSize The largest split header vector size
+   */
+  void parseAndStoreHeaderSections( size_t & largestHeaderVectorSize,
+                                    std::vector< std::vector< string > > & splitHeader );
+
+  /**
+   * @brief Iterate throught the header names vector.
+   * Adjust the size of each header vector by adding empty strings if needed.
+   * Store the modified header names in the corresponding column parameter.
+   * @param largestHeaderVectorSize The largest split header vector size
+   * @param splitHeader A vector containing all split header names
+   */
+  void adjustHeaderSizesAndStore( size_t largestHeaderVectorSize,
+                                  std::vector< std::vector< string > > & splitHeader );
+
+  /**
+   * @brief For each column find and set the column's longest string
+   */
+  void findAndSetMaxStringSize();
+
+  /**
+   * @brief Compute the largest string size in the table. If the table title is the largest string size in the table, recalculate for all
+   * columns the \p m_maxStringSize value by adding extra characters
+   * @param sectionlineLength The length of a section line
+   * @param titleLineLength The length of a title line
+   */
+  void computeAndSetMaxStringSize( string::size_type sectionlineLength,
+                                   string::size_type titleLineLength );
+
+  /**
+   * @brief Compute and build the top and the section line separator
+   * @param topSeparator An empty string to be built
+   * @param sectionSeparator An empty string to be built
+   */
+  void computeAndBuildSeparator( string & topSeparator, string & sectionSeparator );
+
+  /**
+   * @brief Build the table title section
+   * @param titleRows Rows containing the title section.
+   * @param topSeparator The top line separator
+   * @param sectionSeparator The section line separator
+   */
+  void buildTitleRow( string & titleRows, string topSeparator, string sectionSeparator );
+
+  /**
+   * @brief Build a section by specifying it's type ( header or section )
+   * @param sectionSeparator Line separator between sections
+   * @param rows A section row
+   * @param nbRows Indicates the number of lines in a section
+   * @param section The section to be built
+   */
+  void buildSectionRows( string sectionSeparator,
+                         string & rows,
+                         integer const nbRows,
+                         Section const section );
+
+
+  //Vector who contains all rows in the table
+  std::vector< std::vector< string > > m_cellsRows;
+  std::vector< Column > m_columns;
+
+  integer borderMargin;
+  integer columnMargin;
+
+  string tableTitle;
+
+  int marginTitle = 2;
+
+};
+
+template< size_t N, typename ... Args >
+void Table::addRow( Args const &... args )
+{
+  constexpr std::size_t nbColumn_ = sizeof...(args);
+  static_assert( nbColumn_ == N,
+                 "The number of cells per line does not correspond to the number of parameters." );
```

Comment:
N only exist to test if `sizeof...()` works as expected. As far I know, in C++17, you can blindly rely on this syntax, so the `static_assert` and N can be removed.